extensions/applicationinsights-osplugin-js/src/OsPlugin.ts

```diff
@@ -0,0 +1,253 @@
+                super.initialize(coreConfig, core, extensions);
+                let identifier = _self.identifier;
+                _evtNamespace = mergeEvtNamespace(createUniqueNamespace(identifier), core.evtNamespace && core.evtNamespace());
+
+                _os = sessionStorage.getItem("ai_os");
+                _osVer = parseInt(sessionStorage.getItem("ai_osVer"));
+                if(_os && _osVer){
+                    _retrieveFullVersion = true;
+                }
+                _self._addHook(onConfigChange(coreConfig, (details)=> {
+                    let coreConfig = details.cfg;
+                    let ctx = createProcessTelemetryContext(null, coreConfig, core);
+                    _ocConfig = ctx.getExtCfg<IOSPluginConfiguration>(identifier, defaultOSConfig);
+                    _getOSTimeoutMs = _ocConfig.getOSTimeoutMs;
+                    _endpointIsBreeze = _ocConfig.endpointIsBreeze;
+                  
+                    let excludePageUnloadEvents = coreConfig.disablePageUnloadEvents || [];
+
+                    if (_excludePageUnloadEvents && _excludePageUnloadEvents !== excludePageUnloadEvents) {
+                        removePageUnloadEventListener(null, _evtNamespace);
+                        removePageHideEventListener(null, _evtNamespace);
+                        _excludePageUnloadEvents = null;
+                    }
+                   
+                    if (!_excludePageUnloadEvents) {
+                        // If page is closed release queue
+                        addPageUnloadEventListener(_doUnload, excludePageUnloadEvents, _evtNamespace);
+                        addPageHideEventListener(_doUnload, excludePageUnloadEvents, _evtNamespace);
+                    }
+                    _excludePageUnloadEvents = excludePageUnloadEvents;
+                }));
+                function _doUnload() {
+                    _releaseEventQueue();
+                }
+            };
+        
+            _self.processTelemetry = (event: ITelemetryItem, itemCtx?: IProcessTelemetryContext) => {
+                itemCtx = _self._getTelCtx(itemCtx);
+
+                if (!_retrieveFullVersion && !_getOSInProgress && _retryTime < maxRetry) {
+                    // Start Requesting OS version process
+                    _getOSInProgress = true;
+                    // Timeout request if it takes more than 5 seconds (by default)
+                    _getOSTimeout = setTimeout(() => {
+                        _completeOsRetrieve();
+                    }, _getOSTimeoutMs);
+                    startRetrieveOsVersion();
+                    _retryTime++;
+                }
+        
+                if (_getOSInProgress) {
+                    _eventQueue.push({
+                        ctx: itemCtx,
+                        item: event
+                    });
+                } else {
+                    if (_retrieveFullVersion){
+                        if (_endpointIsBreeze){
```

Comment:
For the "setting" of the OS values, you have this here twice, lets extract into a single function